mxcube3/ui/actions/sampleChanger.js

```diff
@@ -97,50 +98,64 @@ export function scan(address) {
   };
 }
 
-export function loadSample(address) {
-  return function (dispatch) {
-    fetch(`mxcube/api/v0.1/sample_changer/mount/${address}`, {
-      method: 'GET',
-      headers: {
-        Accept: 'application/json',
-        'Content-type': 'application/json'
-      },
-      credentials: 'include'
-    }).then(response => {
-      if (response.status >= 400) {
-        dispatch(showErrorPanel(true, response.headers.get('message')));
-        throw new Error(`Error while  sample loading sample @ ${address}`);
-      }
-
-      response.json().then(contents => { dispatch(setContents(contents)); }
-                           ).then(dispatch(refresh()));
-    });
+export function loadSample(sampleData, successCb = null) {
+  return function (dispatch, getState) {
+    const state = getState();
+
+    if (state.sampleChanger.loadedSample.address !== sampleData.location) {
+      fetch('mxcube/api/v0.1/sample_changer/mount', {
+        method: 'POST',
+        credentials: 'include',
+        headers: {
+          Accept: 'application/json',
+          'Content-type': 'application/json'
+        },
+        body: JSON.stringify(sampleData)
+      }).then((response) => {
+        if (response.status >= 400) {
+          dispatch(showErrorPanel(true, response.headers.get('message')));
+          throw new Error('Server refused to mount sample');
+        } else {
+          setCurrentSample(sampleData.sampleID);
+
+          response.json().then(contents => { dispatch(setContents(contents)); }
+                              ).then(dispatch(refresh()));
+
+          if (successCb) {
+            successCb();
+          }
+        }
+      });
+    }
   };
 }
 
-export function unloadSample(address) {
+export function unloadSample(location) {
```

Comment:
is `sampleData` in loadSample arg same datatype as here?
Should we name them the same?